### ISSUE TITLE
Parts of speech triggers

### ISSUE THREAD

**Issue by mariusursache** [silentrob/superscript#120]
Testing rules while writing documentation (using SS Editor):

1. `+ <noun>` matches *I am a cow*, *I am a house*
2. `+ I am a <noun>` fails to match *I am a cow* or *I am a house*

I also have a catch-all wildcard defined, but that should not affect the above behavior

Also, <name> tag is very restrictive, considering all the current immigrants in US. Works with John, Jack, Adam and Jacqueline, but does not work with Jaqueline which is the Latin spelling, or with Marius ;-). What is the source for names?

**Comment by silentrob** [comment]
Here is the code.
https://github.com/silentrob/superscript/blob/master/lib/message.js#L342-L344

Names are derived from ProNouns, and it is a little finicky. I have toyed with adding a name db to prop it up, but it adds more complexity.

**Comment by mariusursache** [comment]
Sorry, I realized my initial comment was missing something, edited it:

1. `+ <noun>` matches *I am a cow*, *I am a house*
2. `+ I am a <noun>` fails to match *I am a cow* or *I am a house*

Any idea why this happens?

**Comment by silentrob** [comment]
So the words are tagged 
```
  taggedWords: 
   [ [ 'I', 'NN' ],
     [ 'am', 'VBP' ],
     [ 'a', 'DT' ],
     [ 'cow', 'NN' ],
```
And the `<noun>` only matches the first noun in the lookup, where `<nouns>` matches all nouns

So...
`+ <noun>` matches *I* ...
`+ I am a <noun>` Will match `I am a I` 
`+ I am a <nouns>` Will match `I am a I` or `I am a cow`
`+ I am a <noun2>` Will match `I am a cow` or `I am a house`

If you think that is unexpected or could be made better, Im open to changing it.

**Comment by mariusursache** [comment]
Ok, so this was a bit confusing because of several things:

1. I expected "I" to be a `<pronoun>`, not also `<noun>`. I did some research and have seen that pronouns are considered a subclass of nouns, which actually makes sense in the anaphora context. I updated the documentation with this specific aspect, to avoid further confusion. However in the tagged words there is no way to differentiate between nouns and pronouns since I=NN and cow=NN are both seen as nouns. Maybe tag pronouns with PN?.
2. The behavior of parts of speech is different than that of other things, and this is again confusing:
 * If `+ I am a <noun>` will not match **I am a boy**, following the logic that `<noun>` is greedier than any word, then
 * `+ I am a *` should not match **I am a boy**, for the same logical reason `*` should be also greedier than any word (which makes no sense). I would suggest `I am a <noun>` to match any of **I am a boy** or **I am a cow**, but will not match **I am alone**. It's more logical I think.
3. Names: a database is hard to compile, since there are tens of thousands of names today (even [this one](http://www.quietaffiliate.com/free-first-name-and-last-name-databases-csv-and-sql/) is far from complete). To avoid inappropriate matching, a name should be considered anything starting with a capital letter, with the recommendation that `<name>` should not be used to catch the first word in a trigger. I think it would still be more useful than a part of speech that does not really match all names.
